### PR TITLE
SI-7439 Avoid NPE in `isMonomorphicType` with stub symbols. …

### PR DESCRIPTION
**Pull request by retronym**
`originalInfo` can return null for stub symbols; deal with that
as we used to before a regression in 016bc3d.

After this change, we can once again delete A_1.class and still compile
code instantiating B_1. (A_1 is only referred to in a method signature
of B_1 which is not called from our code.)

```
scala> new B_1
warning: Class A_1 not found - continuing with a stub.
res0: B_1 = B_1@5284b8f
```

In practice, this situation arises when someone uses a third
party class that was compiled against other libraries not avaialable
on the current compilation classpath.

Review by @paulp

### CODE REVIEW

**Comment by paulp** [comment]
Oops. I wonder what made me think that was safe.

**Comment by retronym** [comment]
Closing for a moment; I need to make the test less fiddly.

**Comment by retronym** [comment]
@paulp Fixed the test, ready for a rubber stamp...

**Comment by retronym** [comment]
Last call for the review train, all aboard!

File: src/reflect/scala/reflect/internal/Symbols.scala [hunk 1]
```diff
@@ -752,7 +752,9 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     final def isMonomorphicType =
       isType && {
         val info = originalInfo
-        info.isComplete && !info.isHigherKinded
+        (    (info eq null)
+          || (info.isComplete && !info.isHigherKinded)
+        )
```
Comment (paulp) [inline src/reflect/scala/reflect/internal/Symbols.scala:757]: Boy, everything is a blur to me now. I'm morally certain I've made this change - more than once! At least my pain is almost at an end - I am well through the process of salvaging or deleting (almost exclusively the latter of course) every branch I have.

**Comment by paulp** [comment]
LGTM

**Comment by paulp** [comment]
Sorry about the inattention.

**Pull request merged**